From af0023f4e05da89fe3e46a1681f11b034fb5f8ab Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Mon, 14 Dec 2020 15:55:39 +0100
Subject: [PATCH 01/19] gmodule: update proc code for linux 5.6+

---
 .../systems/linux/kernel/modules/shared/gmodule.c      | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/sdk-6.5.24/systems/linux/kernel/modules/shared/gmodule.c b/sdk-6.5.24/systems/linux/kernel/modules/shared/gmodule.c
index 0635b811ed8e..55a9ffc6c350 100644
--- a/sdk-6.5.24/systems/linux/kernel/modules/shared/gmodule.c
+++ b/sdk-6.5.24/systems/linux/kernel/modules/shared/gmodule.c
@@ -124,6 +124,15 @@ static int _gmodule_proc_release(struct inode * inode, struct file * file) {
     return single_release(inode, file);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0)
+struct proc_ops _gmodule_proc_fops = {
+    .proc_open =      _gmodule_proc_open,
+    .proc_read =      seq_read,
+    .proc_lseek =     seq_lseek,
+    .proc_write =     _gmodule_proc_write,
+    .proc_release =   _gmodule_proc_release,
+};
+#else
 struct file_operations _gmodule_proc_fops = {
     .owner =      THIS_MODULE,
     .open =       _gmodule_proc_open,
@@ -132,6 +141,7 @@ struct file_operations _gmodule_proc_fops = {
     .write =      _gmodule_proc_write,
     .release =    _gmodule_proc_release,
 };
+#endif
 #else
 int
 gmodule_vpprintf(char** page_ptr, const char* fmt, va_list args)
-- 
2.49.0

