From 432475a51a8a0d15897b0dfe6b24bb00defa06bd Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Mon, 14 Dec 2020 15:55:39 +0100
Subject: [PATCH] gmodule: update proc code for linux 5.6+

---
 systems/linux/kernel/modules/shared/gmodule.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/systems/linux/kernel/modules/shared/gmodule.c b/systems/linux/kernel/modules/shared/gmodule.c
index 9d84166..ba39b96 100644
--- a/systems/linux/kernel/modules/shared/gmodule.c
+++ b/systems/linux/kernel/modules/shared/gmodule.c
@@ -124,6 +124,15 @@ static int _gmodule_proc_release(struct inode * inode, struct file * file) {
     return single_release(inode, file);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0)
+struct proc_ops _gmodule_proc_fops = {
+    proc_open:      _gmodule_proc_open,
+    proc_read:      seq_read,
+    proc_lseek:     seq_lseek,
+    proc_write:     _gmodule_proc_write,
+    proc_release:   _gmodule_proc_release,
+};
+#else
 struct file_operations _gmodule_proc_fops = {
     owner:      THIS_MODULE,
     open:       _gmodule_proc_open,
@@ -132,6 +141,7 @@ struct file_operations _gmodule_proc_fops = {
     write:      _gmodule_proc_write,
     release:    _gmodule_proc_release,
 };
+#endif
 #else
 int
 gmodule_vpprintf(char** page_ptr, const char* fmt, va_list args)
