From 86e62f58bafd848d500b74160429f2345aca4bde Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Mon, 14 Dec 2020 16:02:44 +0100
Subject: [PATCH] kernel-modules: add dropped defines to work with 5.9+

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>

---
 systems/linux/kernel/modules/include/lkm.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/systems/linux/kernel/modules/include/lkm.h b/systems/linux/kernel/modules/include/lkm.h
index 1410294..0b2dab9 100644
--- a/systems/linux/kernel/modules/include/lkm.h
+++ b/systems/linux/kernel/modules/include/lkm.h
@@ -36,6 +36,10 @@
 #include <linux/smp_lock.h>
 #endif
 #include <linux/module.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,9,0)
+#define HAVE_COMPAT_IOCTL 1
+#define HAVE_UNLOCKED_IOCTL 1
+#endif
 
 /* Helper defines for multi-version kernel  support */
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,5,0)
